### PR TITLE
Metadata refactor

### LINKED ISSUE #491: Metadata refactor

### PR DESCRIPTION
**Pull request by muXxer**
- All Metadata features are now `map[string][]byte` instead of `[]byte`
- Removed `StateMetadata` field in favor of new `StateMetadataFeature`

depends on https://github.com/iotaledger/hive.go/pull/600

### CODE REVIEW

File: gen/feat_metadata.tmpl [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// Code generated by go generate; DO NOT EDIT. Check gen/ directory instead.
+
+import (
+	"bytes"
+
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/serializer/v2"
+)
+
+type {{.Name}}Entries map[string][]byte
```
Comment (alexsporn) [inline gen/feat_metadata.tmpl]: There is no mention whatsoever about ordering of the map. This makes serialization non-deterministic.

Comment (alexsporn) [reply gen/feat_metadata.tmpl]: This is now also forcing the use of a key everytime. (e.g. `data` as in every changed testcase)

Comment (alexsporn) [reply gen/feat_metadata.tmpl]: Why is this better than something like IRC-27 or IRC-30? The protocol does not need the map here to function.

Comment (muXxer) [reply gen/feat_metadata.tmpl]: The new serialization of maps is deterministic in serix. So this wouldn't be a problem anymore :+1:

**Review by alexsporn — APPROVED** [review]

**Pull request merged**